File: serpapi_client.py
"""SerpAPI library initialization.

Loads SERPAPI_KEY from .env.local and exposes a ready-to-use client.

Usage:
    from serpapi_client import get_client, SERPAPI_KEY

    client = get_client()
    results = client.search({"engine": "google", "q": "apartments for rent"})
"""

from __future__ import annotations

import os
from pathlib import Path

import serpapi
from dotenv import load_dotenv

# Load .env.local (falls back to .env) from the project root.
_PROJECT_ROOT = Path(__file__).resolve().parent
load_dotenv(_PROJECT_ROOT / ".env.local")
load_dotenv(_PROJECT_ROOT / ".env", override=False)

SERPAPI_KEY: str | None = os.getenv("SERPAPI_KEY")


def get_client() -> "serpapi.Client":
    """Return an initialized SerpAPI client, or raise if the key is missing."""
    if not SERPAPI_KEY or SERPAPI_KEY == "replace_with_your_serpapi_key":
        raise RuntimeError(
            "SERPAPI_KEY is not set. Copy .env.local.example to .env.local and "
            "add your key from https://serpapi.com/manage-api-key"
        )
    return serpapi.Client(api_key=SERPAPI_KEY)

File: serpapiClient.js
// SerpAPI library initialization (Node).
//
// Loads SERPAPI_KEY from .env.local and exposes a configured client.
//
//   import { getJson, SERPAPI_KEY } from "./serpapiClient.js";
//   const results = await getJson({ engine: "google", q: "apartments for rent" });

import { existsSync } from "node:fs";
import { fileURLToPath } from "node:url";
import { dirname, join } from "node:path";
import dotenv from "dotenv";
import { getJson } from "serpapi";

const projectRoot = dirname(fileURLToPath(import.meta.url));

const envLocal = join(projectRoot, ".env.local");
if (existsSync(envLocal)) dotenv.config({ path: envLocal });
dotenv.config({ path: join(projectRoot, ".env"), override: false });

export const SERPAPI_KEY = process.env.SERPAPI_KEY;

if (!SERPAPI_KEY || SERPAPI_KEY === "replace_with_your_serpapi_key") {
  console.warn(
    "[serpapiClient] SERPAPI_KEY is not set. Copy .env.local.example to .env.local " +
      "and add your key from https://serpapi.com/manage-api-key",
  );
}

// Thin wrapper that injects the API key into every request.
export function search(params) {
  return getJson({ ...params, api_key: SERPAPI_KEY });
}

export { getJson };
